File: 2021/python/day2.py


def solvepart1(content):
    coord = [0,0]

    for x in content:
        if x[0] == 'forward':
            coord[0] += x[1]
        elif x[0] == "up":
            coord[1] -= x[1]
        else: 
            coord[1] += x[1]

    return coord[1]*coord[0]

def solvepart2(content):
    coord = [0,0,0]
    
    for x in content:
        if x[0] == 'forward':
            coord[0] += x[1]
            coord[1] += x[1]*coord[2]
        elif x[0] == "up":
            coord[2] -= x[1]
        else:
            coord[2] += x[1]

    return coord[1]*coord[0]


if __name__ == "__main__":

    with open("/Users/hjjoosse/Documents/UMCU/Adventofcode/2021/data/input_d2.txt","r") as f:
        content = f.read().split("\n",)[:-1]
        content = [x.split() for x in content]
        content = [[str(x[0]),int(x[1])]  for x in content]

    print(solvepart1(content))
    print(solvepart2(content))

File: 2021/golang/day2/day2.go
package day2

import (
	"aoc21/utils"
	"fmt"
	"strconv"
	"strings"
)

func SolveDay2() {
	fmt.Println(part1("/Users/hjjoosse/Documents/UMCU/Adventofcode/2021/data/input_d2.txt"))
	fmt.Println(part2("/Users/hjjoosse/Documents/UMCU/Adventofcode/2021/data/input_d2.txt"))
}

func part1(path string) int {
	stringArray := utils.ReadStringArray(path)
	post := [2]int{0, 0}
	for _, s := range stringArray {
		x := strings.Split(s, " ")
		i, err := strconv.Atoi(x[1])
		if err == nil {
			if x[0] == "forward" {
				post[0] += i
			} else if x[0] == "up" {
				post[1] -= i
			} else {
				post[1] += i
			}
		}

	}
	return post[0] * post[1]
}

func part2(path string) int {
	stringArray := utils.ReadStringArray(path)
	post := [3]int{0, 0, 0}
	for _, s := range stringArray {
		x := strings.Split(s, " ")
		i, err := strconv.Atoi(x[1])
		if err == nil {
			if x[0] == "forward" {
				post[0] += i
				post[1] += i * post[2]
			} else if x[0] == "up" {
				post[2] -= i
			} else {
				post[2] += i
			}
		}
	}
	return post[0] * post[1]
}
